.gitlab-ci.yml

```diff
@@ -83,6 +88,21 @@ build-in-docker-static:
   variables:
     EXTRA_CMAKE_FLAGS: "-DBUILD_SHARED_LIBS=OFF"
 
+upload-artifacts:
+  stage: upload
+  extends: .in-prplmesh-builder
+  script:
+    - ln -s "$CI_OWNCLOUD_PASSWORD" ~/.netrc
+    - sed -i 's/^/machine ftp.essensium.com login prplmesh-robot-ci password /' ~/.netrc
+    - ci/upload_to_owncloud.sh -v --overwrite-remote-path "artifacts/$CI_JOB_ID" artifacts/latest ./build
+  rules:
+    - if: $CI_COMMIT_BRANCH == "master"
+  dependencies:
+    - build-in-docker
+    - build-for-netgear-rax40
+    - build-for-glinet-b1300
+    - build-for-turris-omnia
```

Comment:
I think there's a bug in the commit message: "because we don't want the job to start if
an unrelated job from a previous stage failed" -> because we *do* want the job to start if *only* an unrelated job from a previous stage failed?

Reply:
It's not a bug, the commit message correspond to Gitlab CI's behavior.
I wanted the upload to only be done if all the previous job for the pipeline succeeded (so that the `latest` folder on owncloud always corresponds to a build passing all the Gitlab CI tests). That's how downloading the "latest" artifacts using Gitlab API is supposed to work (from the documentation).

Since builds on the master branch are supposed to always be passing anyway, it's arguably not /that/ useful.

Reply:
OK, then maybe "because we don't want the job to start if any test from the previous stage failed". The "unrelated" was confusing.